pkg/data/management/kontainerdriver_data.go

```diff
@@ -75,14 +78,33 @@ func cleanupImportDriver(creator driverCreator) error {
 	return nil
 }
 
+// removeUnsupportedDrivers - remove unsupported drivers
+func removeUnsupportedDrivers(creator driverCreator) error {
+	driverList := []string{"aliyunkubernetescontainerservice", "baiducloudcontainerengine", "huaweicontainercloudengine",
+		"linodekubernetesengine", "opentelekomcloudcontainerengine", "rancherkubernetesengine", "tencentkubernetesengine"}
+
+	var err error
+
+	for _, driver := range driverList {
+		if _, err = creator.driversLister.Get("", driver); err == nil {
+			logrus.Infof("removing kontainer drvier %s", driver)
+			err = creator.drivers.Delete(driver, &v1.DeleteOptions{})
+		}
+		if err != nil && !errors.IsNotFound(err) {
+			return err
```

Comment:
This will be hit if someone upgrades from the rancherVersion with our code to the next. Actually this will be hit if we reboot our rancher pods .  Can you confirm ?

Reply:
this code should execute every time a new Leader is assigned

Reply:
It looks like the code is executed every time a Rancher pod starts.  I can confirm that.  I am not sure about leader assignment changes, I can look into that.

Reply:
the code is only executed when leader election changes